docs/tutorial.md

```diff
@@ -1,87 +1,125 @@
 
-First, decide what language your rule is going to be written in.  I happen to
-know that you could do this in Python, but maybe you've got something else.
+To start simply, create an empty git repo at `/tmp/foo`.  This repo will hold
+the rule and the code the rule is working on.  Of course you can use a different
+path or an existing git repo, just adjust the path examples here.
 
-## Setting up a local development ruleset
+To make the directory realistic enough for ick to run, create an empty
+`pyproject.toml` file and commit it to git.
 
-To avoid getting into the weeds on the many ways to configure it, let's assume
-you have an empty directory `/tmp/foo` (adjust command lines as needed) that you
-want to put your rule in.
+[BTW: it seems odd that this directory has to be a git repo. Why can't ick
+work in a plain-old directory?]
 
-First you'll need to create a user/repo-type config in `ick.toml` in your Python project's root:
+Ick reads `ick.toml` files to find rules.  A ruleset is a location to find
+rules.  In `/tmp/foo` create an `ick.toml` file to say that the current
+directory has rules:
 
 ```toml
 [[ruleset]]
 
 path = "."
 ```
 
-This tells `ick` to look in subdirectories of that for rules, which
-coincidentally can also be put in `ick.toml` in the next step.
+If you run `ick list-rules`, it shouldn't find any:
 
-In this step though, if you run `ick list-rules` it shouldn't find any.
+```shell
+$ ick list-rules
+$
+```
 
 ## Creating a rule definition
 
-Next, we can append to this file and treat it as a rule-type config:
+[TODO: why did we need a ruleset definition if we are going to put explicit rule
+definitions in ick.toml anyway? Maybe `[[ruleset]] path = "."` should be a
+default that always applies.]
+
+Next, we can append to `ick.toml` to define a rule:
 
 ```toml
 [[rule]]
 
 language = "python"
-name = "move_isort_ini"
-scope = "project"
+name = "move_isort_cfg"
+# scope = "project"
 project_types = ["python"]
 ```
 
-This would now run once for each Python project.  This isn't very efficient
-though -- the runner will think this needs to be rerun when *any* file in the
-project changes, when in reality there are just two we care about.  We might read
-both, and might write one and delete the other, so we specify them as both input
-and output:
+The `language` setting means we will implement the rule with Python code.
+Setting `scope` to `project` means that the rule will be invoked on the project
+directory instead of on individual files (but that doesn't work yet, so it's
+commented out).
 
-```toml
-inputs = ["pyproject.toml", "isort.ini"]
-outputs = ["pyproject.toml", "isort.ini"]
-```
+Ick can look for projects of certain types.  Setting `project_types` here means
+the rule will be invoked on projects that ick determines are Python projects.
+
+[TODO: list-rules doesn't error here as the tutorial used to claim: it shows the
+name of the rule. Do we want it to error? It could be helpful to list the rule,
```

Comment:
Agreed, I'd expect it to do some validation.  Not necessarily install and check the deps (or pull a docker image), but validate that the config (including associated local files) is well-formed.